code/modules/gunse/modular_gun_parts.dm

```diff
@@ -426,6 +426,56 @@ ABSTRACT_TYPE(/obj/item/gun_parts/accessory)
 	icon_state = "it_revolver_short"
 	length = 13
 
+/obj/item/gun_parts/barrel/soviet/improvshort
+	name = "Improvised Short Barrel"
+	desc = "Some pipe shrouded with the remains of a soviet spacesuit"
+	spread_angle = -2
+	part_DRM = GUN_FOSS | GUN_SOVIET | GUN_ITALIAN
+	name_addition = "stubby"
+	length = 15
+	icon_state = "improvshortbarrel"
+	icon = 'icons/obj/items/modular_guns/mistfits.dmi'
+
+/obj/item/gun_parts/barrel/soviet/improvlong
+	name = "Improvised Long Barrel"
+	desc = "Several mismatched parts from old rifles, welded together just for you"
+	spread_angle = -4
+	part_DRM = GUN_FOSS | GUN_SOVIET | GUN_ITALIAN
+	name_addition = "rifled"
+	length = 53
+	icon_state = "longimprovbarrel"
+	icon = 'icons/obj/items/modular_guns/mistfits.dmi'
+
+/obj/item/gun_parts/barrel/soviet/improvlong
+	name = "Improvised Long Barrel"
+	desc = "Several mismatched parts from old rifles. Someone painted a trans flag on it"
+	spread_angle = -5
```

Comment:
Ideally the queer variants don't need to be *strictly* better than the normal, but this is fine

Reply:
Made inline with it's parent